tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -41,13 +41,19 @@ spec:
           # oras has very limited support for selecting the right auth entry,
           # so create a custom auth file with just one entry.
           DEST_AUTH_FILE=$(mktemp)
-          select-oci-auth "$4:$5" > "$DEST_AUTH_FILE"
+          registry=$(echo "$4" | cut -d '/' -f 1)
+          select-oci-auth "$4" | jq -c \
+            '.auths."'$4'" = .auths."'$registry'" | del(.auths."'$registry'")' > "$DEST_AUTH_FILE"
```

Comment:
If I'm reading this right, this will basically rename the quay.io key in auths to the full repo name, right? And it's also what a quick google for "jq rename key" gives me, so I guess the answer is yes :)

Reply:
Yep
```
$ echo $somevar
quay.io/foo/bar
$ echo $registry
quay.io
$ ./select-oci-auth.sh quay.io/foo/bar | jq -c '.auths."quay.io/foo/bar" = .auths."'$registry'" | del(.auths."'$registry'")'
Using token for quay.io
{"auths":{"quay.io/foo/bar":{"auth":"sometoken"}}}
```

Reply:
Cool!